### PR TITLE
Swipe to Scan

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/bkbr9fdhrw8bhq3dtfvgeewnwr

[download apk](http://filestorage.bluewallet.io:1488/1577508091.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/rfuzry88fnajp76zke5y4wkjrg

[download apk](http://filestorage.bluewallet.io:1488/1577547418.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/bgk1y1j7ece63e1awzcv6005tw

[download apk](http://filestorage.bluewallet.io:1488/1577554441.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/n29emh6heq4hbj4j3p84x39e68

[download apk](http://filestorage.bluewallet.io:1488/1577555750.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/j4qa9w3m71zgfk5aaeen7md5b4

[download apk](http://filestorage.bluewallet.io:1488/1577559440.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/excetgjpnneykk74jwt3awwc7m

[download apk](http://filestorage.bluewallet.io:1488/1577563696.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/9qk1wunz87jba8zbmhhv0huw5w

[download apk](http://filestorage.bluewallet.io:1488/1577564293.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/jfhhr31gnp9wzkb7u2p68b129w

[download apk](http://filestorage.bluewallet.io:1488/1577565341.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/302bzdjhu6j8eqrn7p39bwz2m4

[download apk](http://filestorage.bluewallet.io:1488/1577570478.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/b1zj1pkr7zn681183g8nmzrua4

[download apk](http://filestorage.bluewallet.io:1488/1577575872.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/0addwzfh47zvchme3rc62f66jw

[download apk](http://filestorage.bluewallet.io:1488/1577579157.apk)

File: class/deeplinkSchemaMatch.js [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    return (
+      lowercaseString.startsWith('bitcoin:') ||
+      lowercaseString.startsWith('lightning:') ||
+      lowercaseString.startsWith('blue:') ||
+      lowercaseString.startsWith('bluewallet:') ||
+      lowercaseString.startsWith('lapp:')
+    );
+  }
+
+  static navigationRouteFor(event, completionHandler) {
```
Comment (Overtorment) [inline class/deeplinkSchemaMatch.js]: several unit tests are required. for all class methods.
also, jsdoc blocks explaining what this method does. at least for `navigationRouteFor`

File: package.json [hunk 2]
```diff
@@ -112,6 +112,7 @@
     "react-native-snap-carousel": "3.8.4",
     "react-native-sortable-list": "0.0.23",
     "react-native-svg": "9.5.1",
+    "react-native-swiper": "git+https://github.com/BlueWallet/react-native-swiper.git#1.5.14",
```
Comment (Overtorment) [inline package.json]: gj. but mI still not 100% convinced that we are safe from malicious deps (deps of deps to be specific), and I dont have an answer for that. I tried googling, but this seems to be an unsolved problem.

File: screen/send/details.js [hunk 3]
```diff
@@ -892,6 +894,7 @@ export default class SendDetails extends Component {
             address={item.address}
             isLoading={this.state.isLoading}
             inputAccessoryViewID={BlueDismissKeyboardInputAccessory.InputAccessoryViewID}
+            launchedBy="SendDetails"
```
Comment (Overtorment) [inline screen/send/details.js]: is there a tidy way to move this to a constant..? maybe class constant..?

Comment (marcosrdz) [reply screen/send/details.js]: Now using this.props.navigation.state.routeName

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/3uq1umqww9cjdp7e3e9qgt8bem

[download apk](http://filestorage.bluewallet.io:1488/1577807226.apk)

File: package.json (hunk 2 above)
Comment (marcosrdz) [reply package.json]: prop-types is the only dependency for this package. There are devDependencies of course.

File: class/deeplinkSchemaMatch.js (hunk 1 above)
Comment (marcosrdz) [reply class/deeplinkSchemaMatch.js]: JSDoc added.

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/wkbewwdck8pm2e0g15cex4b8nw

[download apk](http://filestorage.bluewallet.io:1488/1577822729.apk)

File: class/deeplinkSchemaMatch.js (hunk 1 above)
Comment (marcosrdz) [reply class/deeplinkSchemaMatch.js]: Added unit tests

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/72dem1dftrba7ffx3jvac30d6m

[download apk](http://filestorage.bluewallet.io:1488/1577824510.apk)

File: MainBottomTabs.js [hunk 4]
```diff
@@ -157,6 +160,7 @@ const WalletsStackNavigator = createStackNavigator(
 
 const CreateTransactionStackNavigator = createStackNavigator({
   SendDetails: {
+    routeName: 'SendDetails',
```
Comment (Overtorment) [inline MainBottomTabs.js]: note for future: we should probably add this prop to all screens

File: package.json (hunk 2 above)
Comment (Overtorment) [reply package.json]: Im always stressed about other deps..

File: tests/integration/deepLinkSchemaMatch.test.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    assert.ok(
+      DeeplinkSchemaMatch.isBothBitcoinAndLightning(
+        'bitcoin:1DamianM2k8WfNEeJmyqSe2YW1upB7UATx?amount=0.000001&lightning=lnbc1u1pwry044pp53xlmkghmzjzm3cljl6729cwwqz5hhnhevwfajpkln850n7clft4sdqlgfy4qv33ypmj7sj0f32rzvfqw3jhxaqcqzysxq97zvuq5zy8ge6q70prnvgwtade0g2k5h2r76ws7j2926xdjj2pjaq6q3r4awsxtm6k5prqcul73p3atveljkn6wxdkrcy69t6k5edhtc6q7lgpe4m5k4',
+      ),
+    );
+  });
+
+  it('isLnurl', () => {
+    assert.ok(
+      DeeplinkSchemaMatch.isLnUrl(
```
Comment (Overtorment) [inline tests/integration/deepLinkSchemaMatch.test.js]: all testcases need negative examples as well, like, `isBitcoinAddress('bla')` is NOT an address. but will do for now

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/vnr5d3b54xzxvbmydgkhfm1p14

[download apk](http://filestorage.bluewallet.io:1488/1578170475.apk)

**Pull request merged**